src/main/rx/spektrum.c

```diff
@@ -130,6 +132,51 @@ uint8_t spektrumFrameStatus(void)
 
     rcFrameComplete = false;
 
+	//Check for bind type in the first packet after startup. This is provided in the second byte in a data packet from the internal remote.
+    //This packet wont contain any data if the remote was bound using a separate receiver (bound as external remote)
+    if(!initialPacket){
+
+        switch (spekFrame[1]){
+            case 0xb2:       //11MS 2048 DSMX
+                rxConfig->serialrx_provider = SERIALRX_SPEKTRUM2048;
+                saveConfigAndNotify();
```

Comment:
Ah you're right, I didn't consider flash write cycles. I believe this should work without writing to flash, it would only mean that the setting wouldn't be saved when powered off. That shouldn't matter though if it's always going to check the bind type from the receiver. I'll test this and make sure it works.